File: 10845_queue.py
queue = []

# queue 삽입
def push(component: int) -> None:
    global queue
    queue.append(component)

# queue 빼내기
def pop() -> None:
    global queue
    print(queue[0])
    queue = queue[1:]

N = int(input())

for i in range(N):
    order = input().split(' ')
    if order[0] == 'push':
        intager = order[1]
        order = order[0]
        push(intager)
    elif order[0] == 'pop':
        pop()

File: C_language/10845_queue.c
// https://www.acmicpc.net/problem/10845

#define _CRT_SECURE_NO_WARNINGS

#include <stdio.h>
#include <stdlib.h>

#define MAX_QUEUE_SIZE 10000

typedef struct
{
	int intager;
	queuePointer* ptr;
} queuePointer;

queuePointer* queue = NULL;
int queueSize = 0;

void push(int comp) {
	queuePointer* ptr = NULL;

}

void pop() {
	if (queue == NULL) {
		printf("-1");
	}
	else {
		int temp = queue->intager;
		queue = queue->ptr;
		printf("%d", temp);
	}
}

int main() {
	int N;
	scanf("%d", &N);

	for (int i; i < N; i++) {
		char input[5]; int num;
		scanf("%s", input);

		switch (*input) {
		case 'push':
			scanf("%d", &num);
			push(num);
			break;
		case 'pop':
			pop();
			break;
		}
	}

	return 0;
}